contracts/predeploys.go

```diff
@@ -0,0 +1,20 @@
+package contracts
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/polymerdao/monomer/bindings/generated"
+)
+
+type Predeploy struct {
+	Address          common.Address
+	DeployedBytecode []byte
+}
+
+var L2ApplicationStateRootProviderAddr = common.HexToAddress("0x4e7a96d48e79e61c7aee5ef9e59d7cfc6f0bdc8d")
```

Comment:
Similar: if this is arbitrary, it should probably appear hand-picked.

Reply:
Yeah that's a good idea, the `L2ToL1MessagePasser` contract we predeploy in https://github.com/polymerdao/monomer/pull/86 uses that `0x42000...` Optimism namespace, I'll modify this address to use the same namespace.

Reply:
I updated the tx origin address and the monomer predeploy contract address to use the `0x43000...` to distinguish them from the original OP contracts